src/borg/helpers/manifest.py

```diff
@@ -86,7 +86,10 @@ def list(self, *, glob=None, match_end=r'\Z', sort_by=(), first=None, last=None,
         if isinstance(sort_by, (str, bytes)):
             raise TypeError('sort_by must be a sequence of str')
         regex = re.compile(shellpattern.translate(glob or '*', match_end=match_end))
-        archives = [x for x in self.values() if regex.match(x.name) is not None]
+        if consider_checkpoints:
+            archives = [x for x in self.values() if regex.match(x.name) is not None]
+        else:
+            archives = [x for x in self.values() if ((regex.match(x.name) is not None) and ('checkpoint' not in x.name))]
```

Comment:
well, guess it is a bit a matter of taste, but I'ld rather avoid duplicating the code that would build the full list and just remove the checkpoints from that list. it might be a little slower, but usually we do not have that many archives and the code is simpler that way.

so how about:
```
archives = [x for x in self.values() if regex.match(x.name) is not None]
if not consider_checkpoints:
    archives = [x for x in archives if 'checkpoint' not in x.name]
```

Also please check other places in the code whether `'checkpoint' not in x.name` is the correct way to check for a checkpoint archive name.

Reply:
I did it this way for speed considerations, but I'm happy to modify it to just remove them instead of effectively duplicating if that's preferred. It should be fairly quick in this case I guess so the speed hits wouldn't be significant.

Reply:
The way how you did it is slower as you did not use if/else, but you do build the list twice if consider_checkpoints is False.

with if/else, it would be faster than what I suggested, but I prefer to reduce code duplication.